src/odemis/driver/xt_client.py

```diff
@@ -2402,42 +2398,29 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                     tol_rotation = max(min(movement_req_rotational), STAGE_TOL_ROTATION)  # radians
 
                 axes_to_check = linear_axes_to_check | rotational_axes_to_check
-
-                while True:
-                    current_pos = self.parent.get_stage_position()
-                    # Every axis requested to move should have moved (compared to the position before starting)
-                    # if there is some change w.r.t starting position, proceed to check the target position
-                    # Note that we don't use almost_equal() on the floats, because all we care about is whether
-                    # the position has changed. Even a very tiny change is a sign we've received a position update.
-                    if all(current_pos[a] != op for a, op in orig_pos.items()):
-
-                        if axes_to_check:
-                            axes_updated = isNearPosition(current_pos=current_pos, target_position=target_pos,
-                                                          axes=axes_to_check, rot_axes=rotational_axes_to_check,
-                                                          atol_linear=tol_linear, atol_rotation=tol_rotation)
-                        else:
-                            axes_updated = True  # No axis to check => move is done
-
-                        if axes_updated:
-                            logging.debug("Position has updated fully: from %s -> %s", orig_pos,
-                                          current_pos)
-                            break
-                        else:
-                            logging.debug("Waiting a little longer as position has not updated fully: %s != %s",
-                                          current_pos, target_pos)
-                    else:
-                        logging.debug("Waiting for position to update: %s == %s (for some axes)",
-                                      orig_pos, current_pos)
+                while axes_to_check:
+                    if isNearPosition(current_pos=current_pos, target_position=target_pos,
+                                      axes=axes_to_check, rot_axes=rotational_axes_to_check,
+                                      atol_linear=tol_linear, atol_rotation=tol_rotation):
+                        # Drop axes from the original position, which are not important because they have not moved
```

Comment:
```suggestion
                        # Drop axes from the original position, which are not important because they have not been requested to move
```